cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/NamespaceHttpHandler.java

```diff
@@ -113,8 +114,25 @@ public void create(HttpRequest request, HttpResponder responder, @PathParam("nam
       }
 
       NamespaceConfig config = metadata.getConfig();
-      if (config != null && !Strings.isNullOrEmpty(config.getSchedulerQueueName())) {
-        builder.setSchedulerQueueName(config.getSchedulerQueueName());
+      if (config != null) {
+        LOG.debug("Namespace {} being created with custom configuration {}", metadata.getName(), config);
+        if (!Strings.isNullOrEmpty(config.getSchedulerQueueName())) {
+          builder.setSchedulerQueueName(config.getSchedulerQueueName());
+        }
+        if (!Strings.isNullOrEmpty(config.getRootDirectory())) {
+          if (!new File(config.getRootDirectory()).isAbsolute()) {
+            throw new BadRequestException(String.format("Cannot create the namespace '%s' with the given custom " +
+                                                          "location %s. Custom location must be absolute path.",
+                                                        namespaceId, config.getRootDirectory()));
+          }
+          builder.setRootDirectory(config.getRootDirectory());
+        }
+        if (!Strings.isNullOrEmpty(config.getHbaseNamespace())) {
```

Comment:
unless you don't support updating these properties, then you should have code there warning the user, if they try.